src/ConstructiveSolidGeometry/Transformations.jl

```diff
@@ -1,69 +1,27 @@
-struct ScaledGeometry{T,P<:AbstractGeometry{T}} <: AbstractTransformedGeometry{T}
-    p::P
-    inv_s::SVector{3,T}
-    ScaledGeometry(p::P, s::SVector{3,T}) where {T,P} = new{T,P}(p, inv.(s))
+const Transformations{T} = NamedTuple{(:scale, :rotation, :translation), 
+    Tuple{SVector{3, T}, SMatrix{3, 3, T, 9}, CartesianVector{T}}}
+# Transformation rule:
+#   1. Scale
+#   2. Rotate
+#   3. Translate
+
+transform(g::AbstractPrimitive, t::Transformations) =
+    transform(transform(transform(g, t.scale), t.rotation), t.translation)
+
+transform(p::AbstractVolumePrimitive, v::CartesianVector) = p + v
+transform(p::AbstractVolumePrimitive, r::AbstractMatrix) = r * p
+transform(p::AbstractVolumePrimitive, s::SVector{3}) = s * p
```

Comment:
Do we still want to call these `transform`? We could go for `translate`, `rotate` and `scale` and also change this in L. 9 to make it more clear what is happening (?)